scaffolding: owner input can't handle special characters

**Description**
If the repo owner contains special characters (e.g. `/` & `-`) the scaffolding will dump those into the package name [here](https://github.com/lyft/clutch/blob/98ef0a001adc93b32e0593237af729caa55d0d51/tools/scaffolding/templates/gateway/api/echo/v1/echo.proto#L3) which is invalid.

**Expected Behavior**
The scaffolding should produce a valid package name.

**Actual Behavior**
```bash
info: compiling go
/private/var/folders/56/qnwtygbx6xq8qjhwj55n2nt1zpl4cj/T/clutch-scaffolding-104696746/api/echo/v1
echo/v1/echo.proto:3:13: Expected ";".
make[1]: *** [api] Error 1
2020/08/11 14:24:43 `make api` in the destination dir returned the above error
exit status 1
make: *** [scaffold-gateway] Error 1
```

See this [slack thread](https://lyftoss.slack.com/archives/C015UJ6LED9/p1597174708182900) for more context.

**Version**
<!-- What SHA of Clutch was running? -->
master

**Other Context**
<!-- Include any relevant configuration excerpts, etc. -->

--- a/tools/scaffolding/scaffolder.go
+++ b/tools/scaffolding/scaffolder.go
@@ -19,6 +19,13 @@ const (
 	defaultRepoName              = "clutch-custom-gateway"
 )
 
+// Replace any special characters, that might appear in repository owners or organizations names, but are not supported
+// in proto3 package names.
+var sanitizeProto3Identifier = strings.NewReplacer(
+	"-", "_",
+	"/", "_",
+)
+
 type workflowTemplateValues struct {
 	Name           string
 	PackageName    string
@@ -33,6 +40,11 @@ type gatewayTemplateValues struct {
 	RepoOwner    string
 	RepoName     string
 	RepoProvider string
+}
+
+// SanitizeProto3 identifiers by replacing unsupported characters.
+func (gatewayTemplateValues) SanitizeProto3(s string) string {
+	return sanitizeProto3Identifier.Replace(s)
 }
 
 func promptOrDefault(prompt string, defaultValue string) string {

--- a/tools/scaffolding/templates/gateway/api/echo/v1/echo.proto
+++ b/tools/scaffolding/templates/gateway/api/echo/v1/echo.proto
@@ -1,6 +1,6 @@
 syntax = "proto3";
 
-package {{ .RepoOwner }}.echo.v1;
+package {{ .RepoOwner | .SanitizeProto3 }}.echo.v1;
 
 option go_package = "{{ .RepoProvider}}/{{ .RepoOwner }}/{{ .RepoName}}/backend/api/echo/v1;echov1";
 

